runtime/test/test_argument_conversion.cpp

```diff
@@ -80,14 +100,12 @@ void doTest(mlir::MLIRContext *ctx, std::vector<std::string> &typeNames,
   auto mod = mlir::parseSourceString<mlir::ModuleOp>(code, ctx);
   llvm::outs() << "Source module:\n" << *mod << '\n';
   cudaq::opt::ArgumentConverter ab{"testy", *mod};
-
   // Create the argument conversions
   ab.gen_drop_front(args, startingArgIdx);
-
-  // Dump the conversions
-  llvm::outs() << "========================================\n"
-                  "Substitution module:\n"
-               << ab.getSubstitutionModule() << '\n';
+  // Dump the modified source module
+  llvm::outs() << "Source module (after):\n" << *mod << '\n';
```

Comment:
Is this dumping the same thing? It looks like it was dumping the substitution module but now is dumping the module receiving the substitutions?

Reply:
Since this is testing _argument conversions_, we must care about building the substitutions properly. The code into which they are applied is only interesting at the point of _argument synthesis_, no?